### DISCUSSION TITLE
Custom Domain - CNAME and CAA conflict resolved with A record, it works, but now DNS check is unsuccessful

### DISCUSSION THREAD

**Discussion by leifhanack** [community/community#42997] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi there, 
we have a GH page (http://curly-fiesta-ff240ffa.pages.github.io) with a custom domain [tech.europace.de](https://tech.europace.de).

Initially we created in our DNS a CNAME tech that pointed to `curly-fiesta-ff240ffa.pages.github.io.`
It results in the following error when GitHub did the DNS check:

> We encountered the following errors attempting to issue a certificate:
> 
> Taking a look at the Certification Authority Authorization (CAA) records, at least one CAA record must exist with the value letsencrypt.org for your site.
>  
> 
> In order for GitHub to automatically provision an SSL certificate you'll need to include a CAA record with the value 0 issue "letsencrypt.org".

We had the following CAA records for `europace.de` in place but it seemed they couldn't be "found". 
<img width="921" alt="image" src="https://user-images.githubusercontent.com/128227/210140470-fa239c74-4f02-484e-820c-050d12a9f5ee.png">

Next, we tried to create a CAA records for the subdomain `tech` but our DNS said it is not possible to have a CNAME record `tech` and also a CAA record `tech`. 

We ended up with changing the CNAME `tech` into an A record and having CAA records for the subdomain `tech`. 

<img width="1228" alt="image" src="https://user-images.githubusercontent.com/128227/210140615-2a90ccaf-42a5-42ce-9788-051a326a3a1c.png">
<img width="817" alt="image" src="https://user-images.githubusercontent.com/128227/210140605-1e83091e-3308-487d-ba4a-15ce6e35bd64.png">

That works but unfortunately the GitHub DNS check is now complaining with `DNS check unsuccessful`

<img width="1351" alt="image" src="https://user-images.githubusercontent.com/128227/210140638-29b5e908-8276-4a65-9271-0a74612cb00c.png">

Do you have an idea what the problem is and how we should set up the records so that we can use a CNAME `tech` and have valid CAA records?

Thanks in advance, Leif

**Comment by yoannchaudet** [comment]
Domains on `pages.github.io` are ephemeral. Like the warning is explaining in your screenshot above, we recommend you to setup a `CNAME` record pointing to `<org or user name>.github.io`. If your domain is an APEX one, [you can use `A` (and `AAAA` if you like) records instead](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain).

Also to confirm, I don't think there is a problem with having both a `CNAME` and `CAA` record for `tech`. These are different record types.

**Comment by leifhanack** [comment, marked as answer]
Thank you Yoann,
 
our solution was:

- we couldn't have a CNAME and CAA record with the same name
- we removed all subdomain CAA records
- we kept all our top level CAA records
- we changed our ALIAS back to a CNAME and pointed it to `europace.github.io.` instead of `curly...`

GH could then load our top level CAA records and the CNAME also worked now.

Thanks, Leif

**Reply by QScienceC** [reply]
@leifhanack , I'm having the same issue, but I don't fully understand your response. Is it possible to see what your CNAME, CAA , and ALIAS files look like?

**Reply by leifhanack** [reply]
Sure.

Top level CAA records:

<img width="694" alt="image" src="https://user-images.githubusercontent.com/128227/226874765-5cae8e07-1799-4c47-b473-b3bd562b6a2a.png">

CNAME:
<img width="686" alt="image" src="https://user-images.githubusercontent.com/128227/226875079-18f51746-2292-4fa8-b603-395b769eb8a9.png">

GitHub pages setup:
<img width="781" alt="image" src="https://user-images.githubusercontent.com/128227/226876497-1b630024-103c-4cff-a747-3384428cc0ac.png">

**Reply by QScienceC** [reply]
Thank you! I got this working now! I did not need the CAA records in google domains, but I did the the CNAME there (I thought I only needed it in my repo).
